.azure-pipelines/gpu-tests.yml

```diff
@@ -29,11 +29,10 @@ jobs:
     container:
       # base ML image: mcr.microsoft.com/azureml/openmpi3.1.2-cuda10.2-cudnn8-ubuntu18.04
       # run on torch 1.8 as it's the LTS version
-      # TODO: Unpin sha256
-      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.7-torch1.8@sha256:b75de74d4c7c820f442f246be8500c93f8b5797b84aa8531847e5fb317ed3dda"
+      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.7-torch1.11"
```

Comment:
As the comment says, we choose 1.8 because its PyTorch's LTS version.

IMO 1.8 should be kept until there's a new LTS (1.12?)

Reply:
The only reason why I'm opposed to testing only with 1.8 is that we cannot test new features that are available in newer PyTorch versions. (For example, the tests fixed in #12878 were never valid since their addition. Luckily, it didn't require any changes to the codebase though.)

I think there're a few options here:
1. run tests with 1.8 (as is)
2. run tests with 1.11
3. run tests with 1.8 **+ run only tests that require 1.{9,10,11} with 1.11**

For the above reason I mentioned, I'm in favour of 2. which is the simplest, but we can try 3. if we really want to keep testing with LTS. (To be precise, the current version we're using in GPU CI is `1.8.1`, so we need to update it to the actual LTS version `1.8.2` if we do.)

@carmocca @Borda Would love to hear your thoughts!

Reply:
I am thinking about running both 1.8 and 1.11

Reply:
As suggested by @Borda, we'll have both 1.8 and 1.11, and this PR will now focus on unpinning the docker image without updating it to PyTorch 1.11.